front/src/applications/operationalStudies/helpers/upsertNewProjectedTrains.ts

```diff
@@ -17,11 +20,16 @@ const upsertNewProjectedTrains = (
   for (const [trainIdKey, trainData] of projectedTrainsToUpsert) {
     const matchingTrain = trainSchedulesById.get(trainIdKey);
     const projectedTrain = {
-      id: trainIdKey,
+      id: isPacedTrain(trainIdKey)
+        ? formatEditoastTrainIdToOccurrenceId({ pacedTrainId: trainIdKey, occurrenceIndex: 0 })
+        : trainIdKey,
       name: matchingTrain?.train_name || 'Train name not found',
       departureTime: new Date(trainData.departure_time),
       spaceTimeCurves: trainData.space_time_curves,
       signalUpdates: trainData.signal_updates,
+      paced: isPacedTrain(trainIdKey)
```

Comment:
You can use `isPacedTrainResponseWithPacedTrainId`, and you can remove the cast
```suggestion
      paced: isPacedTrainResponseWithPacedTrainId(matchingTrain)
```

Reply:
oh thanks i did'nt know about this utils